### PR TITLE
ROX-19958: Adjust Fast Stream builds to be manifest-deployable

### PR DESCRIPTION
**Pull request by msugakov**
## Description

This also addresses ROX-22338.

## Checklist
- [x] Investigated and inspected CI test results

These aren't relevant and won't be done:
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

### Here I tell how I validated my change

- [x] I verified that E2E tests have no regressions caused by this.
- [x] Deployed the product with manifests without image repo/tag overrides and got a functioning product (no smoke test, just cluster and system status).

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by msugakov** [comment]
Manual testing round.

```bash
# generate Central bundle
$ podman run --rm -it -u 0:0 -v ./:/tmp -e ROX_ROXCTL_IN_MAIN_IMAGE=false quay.io/rhacs-eng/roxctl:4.4.x-800-g4beec9ed6d-fast central generate openshift pvc --output-dir /tmp/central-bundle --enable-pod-security-policies=false
$ cd central-bundle

# usual Central deployment stuff
$ central/scripts/setup.sh
$ oc create -R -f central
$ scanner/scripts/setup.sh
$ oc create -R -f scanner
$ scanner-v4/scripts/setup.sh
$ oc create -R -f scanner-v4

# in parallel tab
$ ks port-forward deploy/central 8000:8443

# download roxctl from UI using admin password

# generate Secured Cluster bundle
$ cd ..
$ mv ~/Downloads/roxctl ./
$ chmod +x roxctl
$ ROX_ADMIN_PASSWORD="Btcusa75HuOZePkjPXU6ChnCU" ./roxctl sensor generate openshift --enable-pod-security-policies=false -e localhost:8000 --insecure-skip-tls-verify --name misha-test

# usual Secured cluster deployment
$ cd sensor-misha-test
$ ./sensor.sh

# here's what got deployed
$ ks get po -o yaml | grep 'image:' | sort | uniq
      image: quay.io/rhacs-eng/central-db:4.4.x-800-g4beec9ed6d-fast
      image: quay.io/rhacs-eng/collector:3.18.x-265-g09e782ef16-fast-slim
      image: quay.io/rhacs-eng/main:4.4.x-800-g4beec9ed6d-fast
      image: quay.io/rhacs-eng/scanner:2.33.x-64-gc45794f8f1-fast
      image: quay.io/rhacs-eng/scanner-db:2.33.x-64-gc45794f8f1-fast
    - image: quay.io/rhacs-eng/scanner-db:2.33.x-64-gc45794f8f1-fast
      image: quay.io/rhacs-eng/scanner-slim:2.33.x-64-gc45794f8f1-fast
      image: quay.io/rhacs-eng/scanner-v4:4.4.x-800-g4beec9ed6d-fast
      image: quay.io/rhacs-eng/scanner-v4-db:4.4.x-800-g4beec9ed6d-fast
```

**Comment by msugakov** [comment]
`*-sensor-profiling` failed because it's WIP https://github.com/stackrox/stackrox/pull/7906.

`ocp-4-candidate-qa-e2e-tests` has a [failure](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/stackrox_stackrox/11232/pull-ci-stackrox-stackrox-master-ocp-4-candidate-qa-e2e-tests/1795391756808753152) in a particular test (`NodeInventoryTest: Verify node inventories and their scans`), not caused by this PR.

`ocp-4-12-ui-e2e-tests` and `gke-race-condition-qa-e2e-tests` need to be retriggered.

File: image/rhel/konflux.Dockerfile [hunk 1]
```diff
@@ -40,10 +40,11 @@ COPY . .
 RUN git restore scripts/konflux/bootstrap-yarn/package-lock.json && \
     scripts/konflux/fail-build-if-git-is-dirty.sh
 
+ARG VERSIONS_SUFFIX
+ENV MAIN_TAG_SUFFIX="$VERSIONS_SUFFIX" COLLECTOR_TAG_SUFFIX="$VERSIONS_SUFFIX" SCANNER_TAG_SUFFIX="$VERSIONS_SUFFIX"
+
 ENV GOFLAGS=""
 ENV CGO_ENABLED=1
-# TODO(ROX-19958): figure out if we need BUILD_TAG
```
Comment (tommartensen) [inline image/rhel/konflux.Dockerfile:45]: Since you removed this comment, is it "figured out"?

Comment (msugakov) [reply image/rhel/konflux.Dockerfile:45]: For me - yes. Please ask a question if you'd like me to clarify anything.

Comment (tommartensen) [reply image/rhel/konflux.Dockerfile:45]: Yes, how does the image know its own version? 

Screenshot 2024-05-29 at 14 05 20

Comment (tommartensen) [reply image/rhel/konflux.Dockerfile:45]: Clarified on a call that this version comes from `status.sh`.

**Review by tommartensen — APPROVED** [review]

**Pull request merged**